source/MaterialXCore/Node.cpp

```diff
@@ -718,7 +719,8 @@ bool NodeGraph::validate(string* message) const
                 const string& interfaceName = input->getInterfaceName();
                 if (!interfaceName.empty())
                 {
-                    InputPtr interfaceInput = input->getInterfaceInput();
+                    ValueElementPtr interfaceElement = input->getInterfaceElement();
+                    InputPtr interfaceInput = interfaceElement ? interfaceElement->asA<Input>() : nullptr;
```

Comment:
It seems this cast to an InputPtr is done at every use of the getInterfaceElement() method. Would it make sense to always return an InputPtr instead of a ValueElementPtr? Or is there some specialization of this method that will return something else?

Reply:
This used to be `getInterfaceInput()` but I changed this to a virtual override of `getInterfaceElement()`. This so I could use `getInterfaceElement()` from the modified getValues() call. 

I could keep the original method to avoid this which would also retain API compatibility.

Reply:
Ah I see. Would it make sense to instead let `getValue()` be a virtual method, and override it on the `Input` class to look for interface connections? The we avoid always looking for interface connections, and the method `getInterfaceInput()` could be unchanged, I think..

Reply:
Yes this does seem like a better approach. Based on the latest feedback from Jonathan, we can go this way but probably only add this to getActiveValue() as getValue() has too many possible side-effects and getResolvedValue() isnt' meant to perform traversal. I'll wait until Jonathan's PR is in then revisit this.